test/Streamly/Test/Data/Unbox.hs

```diff
@@ -37,43 +59,143 @@ data NestedSOP =
        NSOP0 SumOfProducts
      | NSOP1 SumOfProducts deriving (Unbox, Show, Generic, Eq)
 
-test ::(Unbox a, Eq a) => a -> a -> Property
-test v1 v2 = monadicIO $ do
-    ref <- run $ IORef.newIORef v1
-    x1 <- run $ IORef.readIORef ref
-    run $  IORef.writeIORef ref v2
-    x2 <- run $  IORef.readIORef ref
-    assert (x2 == v2)
-    assert (x1 == v1)
+-- Write v1 to an Unboxed ref, read it back and compare.
+-- Then do the same for v2.
+test ::(Unbox a, Eq a, Show a) => a -> a -> IO ()
+test v1 v2 = do
```

Comment:
We are testing `v1` independent of `v2`.
Why do we have both `v1` and `v2` here?

Reply:
I guess this is to use the same IORef to read/write two different values.

Reply:
Are we testing `IORef`?